### PR TITLE
[spi_device] Add mbist_en port

### PR DESCRIPTION
**Pull request by eunchan**
SPI_DEVICE Dual port SRAM receives muxed clock. The expected clock
frequencies are 24MHz (generic mode) and 48MHz (SPI Flash mode).

But the flash mode needs an external clock, SPI_SCK, which is the faster
clock 48MHz.

For the MBIST, it should use the faster clock to test the SPI_DEVICE
internal DPSRAM. This commit uses SCAN_CLOCK as a MBIST clock, as SCAN
clock is 48MHz at this time.

### CODE REVIEW

**Review by msfschaffner — APPROVED** [review]
LGTM, but let's wait for @arnonsha to approve as well.

File: hw/ip/spi_device/rtl/spi_device.sv
Comment (eunchan) [reply hw/ip/spi_device/rtl/spi_device.sv]: Oops. Thanks ! fixed!

Comment (arnonsha) [reply hw/ip/spi_device/rtl/spi_device.sv]: @Leonid-Shamis
Looks good Eunchan. We will try it out tomorrow morning and update.

**Comment by eunchan** [comment]
Just updated TB too. Let's wait until the check passed and merge.

File: hw/ip/spi_device/rtl/spi_device.sv
Comment (msfschaffner) [reply hw/ip/spi_device/rtl/spi_device.sv]: Good catch, thanks @arnonsha.

**Comment by arnonsha** [comment]
@eunchan 
After trying it out in Tessent, it seems like we also need set the mux selector in clk_io_div2
Can you please also add mbist_en_i input to prim_clk_div and connect it to const 0 as well ?
  
  // make sure selection point is away from both edges
    prim_clock_mux2 #(
      .NoFpgaBufG(1'b1)
    ) u_step_down_mux (
      .clk0_i(q_p),
      .clk1_i(clk_i),
      .sel_i(step_down_nq & **~mbist_en_i**),
      .clk_o(clk_int)
    );

The cell is also used for the clk_io_div4 but there this logic will be optimized and removed by DC, as it will remain const 0.

**Pull request merged**
